magda-authorization-api/src/test/buildCreateApiRouter.spec.ts

```diff
@@ -0,0 +1,505 @@
+import {} from "mocha";
+import * as sinon from "sinon";
+import * as request from "supertest";
+import * as express from "express";
+import * as yargs from "yargs";
+import addJwtSecretFromEnvVar from "@magda/typescript-common/dist/session/addJwtSecretFromEnvVar";
+import buildJwt from "@magda/typescript-common/dist/session/buildJwt";
+import createApiRouter from "../createApiRouter";
+import { expect } from "chai";
+import jsc from "@magda/typescript-common/dist/test/jsverify";
+import mockDatabase from "./mockDatabase";
+import mockUserDataStore from "@magda/typescript-common/dist/test/mockUserDataStore";
+import Database from "../Database";
+import { userDataArb } from "./arbitaries";
+import { Request } from "supertest";
+import { User } from "@magda/typescript-common/src/authorization-api/model";
+
+describe("Auth api router", function(this: Mocha.ISuiteCallbackContext) {
+    this.timeout(10000);
+
+    let app: express.Express;
+    let argv: any;
+    let newlyCreatedUsers: User[] = [];
+
+    before(function() {
+        argv = retrieveArgv();
+        app = buildExpressApp();
+    });
+
+    function retrieveArgv() {
```

Comment:
I think it might be a good idea as I also felt the similar code typed twice ;-)
Do you prefer it (and commonYargs.ts) to `typescript-common` or move to `magda-authorization-api` as different modules might requires different parameters?

Reply:
I'm happy for `commonYargs` to stay where it is - just letting you know about `fakeArgv.ts` because it allows you to make an argv type from an object, so you don't have to include all the other stuff you don't need for tests like descriptions and so forth - like:

```typescript
const args: argv.Arguments = fakeArgv({
    listenPort: 6104,
    dbHost: "localhost"
    //etc
})
```

Reply:
Do think it's good to make `fakeArgv.ts` more generic like:
```
import { Arguments } from "yargs";

export default function fakeArgv(options: {
    internalUrl?: string;
    registryUrl?: string;
    jwtSecret?: string;
    userId?: string;
    listenPort?: number;
    dbHost?: string;
    dbPort?: string;
}): Arguments {
    return {
        ...options,
        $0: "",
        _: []
    };
}
``` 
And then put it to `magda-typescript-common`?
We might be able to re-use it from there in future.

Reply:
Yep - I'm pretty sure you could just make it `fakeArgv(options: any)`

Reply:
That's a good idea~ so we don't have to keep updating the type definition of `options`.
I've added new commit for this.